docs/docs/security.md

```diff
@@ -1,167 +1,162 @@
 
-Because side channel attacks present tradeoffs between privacy and performance, analysing them requires a different 
-mindset to normal security analysis. Formally, we can say a secret has been leaked via a side channel if we can 
-learn even a single binary bit of data about an encrypted message. However often the leaked data doesn't matter or 
-isn't a secret worth protecting. In these cases it may be better to allow a limited amount of leakage to obtain
-better performance. Future versions of this guide and the Conclave API will assist you in studying and making these
-tradeoffs.
-
-A standard example is whether a message is of type A or type B, under the assumption that most apps have at least 
-two kinds of message that an enclave can process from clients. Careful requirements analysis may reveal that
-the number or sequencing of message types doesn't reveal any important information and thus doesn't need to be 
-protected: only the specific data within the messages.      
-
 ### Architectural attacks
 
-Architectural side channel attacks exploit the nature, structure or specific code of your application architecture to 
-reveal secrets. 
-  
+Architectural side-channel attacks exploit the nature, structure, or specific code of your application architecture to
+reveal secrets.
+
 Here is a non-exhaustive set of examples:
 
-* **Message sizes**. If your enclave is known to process only two kinds of message of around 1 kilobyte or 100 kilobytes
-  respectively, then the size of the encrypted message by itself leaks information about what kind of message it is.
-* **Message processing time**. The same as message sizes but with processing time, e.g. a message type that takes 1 millisecond 
-  to process vs 100 milliseconds can leak what kind of message it is by simply observing how much work the enclave does when
-  the host passes it the new data.
-* **Storage access patterns**. If your enclave doesn't access the database when processing a message of type A, but does
-  when processing a message of type B, or accesses the database with a different sequence, number or type of accesses,
-  the host can learn the message type by observing those accesses.  
+* **Message sizes**. If an enclave is known to process only two kinds of message of around 1 kilobyte or 100 
+  kilobytes, then the size of the encrypted message by itself leaks information about what kind of message it is.
+* **Message processing time**. If an enclave processes a message type in 1 millisecond and another message type in 
+  100 milliseconds, observing how much work the enclave does for a new message, can reveal what kind of message it is.
+* **Storage access patterns**. Suppose an enclave doesn't access the database when processing a message of type A, 
+  but does when processing a message of type B, or accesses the database with a different sequence, number or type 
+  of accesses. In that case, the host can learn the message type by observing those accesses.
```

Comment:
Add that Conclave randomises the sector accesses when the enclave uses the persistence file system to reduce this side channel for the user. It's mentioned in persistence.md.

Reply:
Added.